datafusion/core/src/franz_sinks/kafka_sink.rs

```diff
@@ -58,23 +83,23 @@ impl KafkaSink {
 #[async_trait]
 impl FranzSink for KafkaSink {
     async fn write_records(&mut self, batch: RecordBatch) -> Result<(), DataFusionError> {
-        if batch.num_rows() > 0 {
-            let topic = self.config.topic.as_str();
-
-            let buf = Vec::new();
-            let mut writer = LineDelimitedWriter::new(buf);
-            writer.write_batches(&vec![&batch]).unwrap();
-            writer.finish().unwrap();
-            let buf = writer.into_inner();
-
-            let record = FutureRecord::<[u8], _>::to(topic).payload(&buf);
-                    // .key(key.as_str()),
-
-            let _delivery_status = self
-                .producer
-                .send(record, Duration::from_secs(0))
-                .await;
-        }
+        let topic = self.config.topic.as_str();
+
+        let buf = Vec::new();
+        let mut writer = LineDelimitedWriter::new(buf);
+        writer.write_batches(&vec![&batch]).unwrap();
+        writer.finish().unwrap();
+        let buf = writer.into_inner();
+
+        let record = FutureRecord::<[u8], _>::to(topic).payload(&buf);
+        // .key(key.as_str()),
+
+        let _delivery_status = self
+            .producer
+            .send(record, Duration::from_secs(0))
+            .await
+            .expect("Message not delivered");
+
         Ok(())
```

Comment:
why not pass the Result form the await itself?

Reply:
no reason, that's a good idea